app/models/virtual_cap_pool.rb

```diff
@@ -0,0 +1,41 @@
+class VirtualCapPool < ApplicationRecord
+  belongs_to :responsible_body
+  has_many :school_virtual_caps, dependent: :destroy
+  has_many :school_device_allocations, through: :school_virtual_caps
+  has_many :schools, through: :school_device_allocations
+
+  enum device_type: {
+    'coms_device': 'coms_device',
+    'std_device': 'std_device',
+  }
+
+  def recalculate_caps!
+    self.cap = school_device_allocations.sum(:cap)
+    self.devices_ordered = school_device_allocations.sum(:devices_ordered)
+    save!
+    # TODO: trigger events for CC here?
```

Comment:
I wonder whether _any_ change on this model should trigger update events to CC?

Reply:
I'm trying to figure out the one/many-to-many relationships here.... as far as i can see, a RB can have many VirtualCapPools because it would have one per device type, but no more than 2. Is that right?

If that's the case, then should this model have a uniqueness validation, along the lines of:
`validates :device_type, uniqueness: { scope: [:responsible_body_id] }`
to guard against an RB possibly having multiple pools per device_type ?

Reply:
Yes, good call